integration/tests/basic/rpc/test_rpc_estimate_gas.py

```diff
@@ -0,0 +1,173 @@
+import typing as tp
+
+import allure
+import pytest
+
+from integration.tests.basic.helpers import rpc_checks
+from integration.tests.basic.helpers.basic import BaseMixin
+from utils import helpers
+
+
+@pytest.fixture(scope="class")
+def common_contract(web3_client, class_account) -> tp.Any:
+    common_contract, _ = web3_client.deploy_and_get_contract(
+        "common/Common", "0.8.12",
+        contract_name="Common", account=class_account
+    )
+    yield common_contract
+
+
+@allure.feature("JSON-RPC-ESTIMATE-GAS validation")
+@allure.story("Verify eth_estimateGas RPC call")
+class TestRpcEstimateGas(BaseMixin):
+    account: "eth_account.signers.local.LocalAccount" = None
+
+    @pytest.fixture(params=[(850000, 15000), (8500000, 150000), (8500000, 150000)])
+    def constructor_args(self, request: tp.Any) -> tp.List[int]:
+        return request.param
+
+    @pytest.fixture(params=["BigGasFactory1", "BigGasFactory2"])
+    def deploy_big_gas_requirements_contract(
+            self, request: tp.Any, constructor_args: tp.List[int]
+    ) -> "web3._utils.datatypes.Contract":
+        """Deploy contracts"""
+        self.account = self.sender_account
+        #  contract
+        contract_interface = helpers.get_contract_interface(
+            contract="issues/Ndev49", version="0.8.10", contract_name=request.param
+        )
+        counter = self.web3_client.eth.contract(
+            abi=contract_interface["abi"], bytecode=contract_interface["bin"]
+        )
+        # Build transaction
+        transaction = counter.constructor(*constructor_args).build_transaction(
+            {
```

Comment:
i don't understand what you do here? Why you execute constructor manually and deploy contract manually?
Make some asserts in fixture it's a bad pattern, fixture should return only object for test. You could simplify it to simple deploy

Reply:
This code was just moved from the `test_rpc_base_calls` class `as is` to keep everything related to the gas estimation feature in one place. So all those questions could be directed to the original author.
I will rewrite it, but it wasn't mentioned in the original task goals.

Reply:
Yes, but we need improve our tests too

Reply:
Rewritten